rpc/client.go

```diff
@@ -106,15 +107,38 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 		Closed:       make(chan struct{}),
 		clock:        context.localClock,
 		remoteClocks: context.RemoteClocks,
-		cached:       !context.DisableCache,
 	}
-	if !context.DisableCache {
-		clients[c.Addr().String()] = c
+	clients[c.Addr().String()] = c
+
+	tlsConfig, tlsErr := context.GetClientTLSConfig()
```

Comment:
you might as well error out right away (well, log an error), if tlsErr is not nil, we shouldn't be retrying.